pygac/reader.py

```diff
@@ -894,7 +894,7 @@ def correct_scan_line_numbers(self):
                    'n_orig': self.scans['scan_line_number'].copy()}
 
         # Remove scanlines whose scanline number is outside the valid range
-        within_range = np.logical_and(self.scans["scan_line_number"] < 15000,
+        within_range = np.logical_and(self.scans["scan_line_number"] < len(self.scans) * 2,
```

Comment:
Oh, that was me, sorry... 🙈 But I think the proposed solution would exclude everything, if the number of scanlines is small (e.g. 500), but the scan line number has a large offset (e.g. 2000, 2001, 2002, ...). Probably it's impossible to deduce a valid range for the scanline number from the dimensions of the data. What about a dictionary `{data_type: valid_range}`?